Demo/Demo/UseCases/GetBillingAgreementToken.swift

```diff
@@ -0,0 +1,38 @@
+import UIKit
+
+final class GetBillingAgreementToken {
+
+    func execute() async throws -> Order {
+        return try await DemoMerchantAPI.sharedService.createOrder(order: GetBillingAgreementToken.baTokenRequest)
+    }
+
+    private static let baTokenRequest = """
+        {
```

Comment:
Here's a [good site](https://www.hackingwithswift.com/read/7/3/parsing-json-using-the-codable-protocol) that talks about JSON Parsing in swift:
> Swift has built-in support for working with JSON using a protocol called Codable. When you say “my data conforms to Codable”, Swift will allow you to convert freely between that data and JSON using only a little code.

It's common practice to use `Codable` for encoding/decoding JSON, like we do [here](https://github.com/paypal/iOS-SDK/blob/main/Demo/Demo/Models/CreateOrderParams.swift) in `CreateOrdersParams`. 

It also then allows us to separate out our "business logic" of determining what data to populate our JSON request with, from the model layer ([see BaseViewModel](https://github.com/paypal/iOS-SDK/blob/4ef75dcf25609846445dc5bd2eac70b975a41798/Demo/Demo/ViewModels/BaseViewModel.swift#L84-L88)).

Reply:
This looks like the same request as in [CreateOrderParams.swift](https://github.com/paypal/iOS-SDK/blob/main/Demo/Demo/Models/CreateOrderParams.swift), but with additional values.

We can extend the existing model with optional params, something like:

```
struct CreateOrderParams: Codable {
    let intent: String
    let purchaseUnits: [PurchaseUnit]
    let applicationContext: ApplicationContext
    let paymentSource: PaymentSource?
}

struct PurchaseUnit: Codable {
    let amount: Amount
}

struct Amount: Codable {
    let currencyCode: String
    let value: String
}

struct ApplicationContext: Codable {
    let returnUrl: String
    let cancelUrl: String
    let shippingPreference: String?
}

struct PaymentSource: Codable {
    let paypal: Attributes
}

struct Attributes: Codable {
    let vault: Vault
}

struct Vault: Codable {
    let confirmPaymentToken: String
    let usageType: String
}
```

Reply:
Also, `struct`s are pass by value (instead of `class`es pass by reference) and are preferred in Swift for data/model types as they're thread safe. They're also lighter weight and faster than classes in Swift. Here's a [good read ](https://www.avanderlee.com/swift/struct-class-differences/)on it.

Reply:
I did it as string because its always going to be static values, and this json is super speficic for this specific flow. Do we want to create all of these structs with static data which are only going to be used in one place?

Reply:
I understand - this PR opted for static JSON as its values wont change anywhere within the demo. I think we still want to use `Codable` for a few reasons:
* It's best practice for encoding/decoding JSON in swift & more closely resembles a real merchant integration
* Allows us to define our models in 1 consistent way throughout the demo
* Allows us the flexibility to modify params if needed in the future + provides encoding/decoding error handling